### ISSUE TITLE
Handle async design function

### ISSUE THREAD

**Issue by runemadsen** [designsystemsinternational/mechanic#143]
**Is your feature request related to something something in Mechanic? Please describe.**
We should support `async` design functions

**Is your feature request related to a problem? Please describe.**
It actually works, but the canvas seems to be rendered multiple times when updating the values
<img width="836" alt="Screen Shot 2022-09-02 at 2 31 31 PM" src="https://user-images.githubusercontent.com/192021/188143489-f29505c7-c3b7-4a6b-a7c8-9040ee3fc2ec.png">

**Describe the solution you'd like**
That I can write a design function that uses `async` and it just works. This would not be something that could be used for `engine-react` since React components are not allowed to be async.

**Comment by lucasdinonolte** [comment]
Seeing this – might not be 100% related to the `async` question – made me think about the image input: Would it make sense to move the async image loading to that input? So a design function is always passed a base64 image it can use immediately? 

I think this would make design functions a bit cleaner as it would remove user code to turn the FileBlob into an image they can work with.

**Comment by lucasdinonolte** [comment]
I just tried the above approach and this would actually lead to exceeding the local storage quota really fast 📦🥸

**Comment by fdoflorenzano** [comment]
Yeah that's the original reason we didn't do it that way. It somehow needs to involve uploading the image somewhere and keeping a reference to it. We've discussed a couple of times in the past about maybe adding a config prop that gives the auth credentials for an S3 bucket or Dropbox folder to make this happen as an option for better image handling.
